### ISSUE TITLE
build issue

### ISSUE THREAD

**Issue by jbmohler** [ockam-network/ockam#1201]
I have an initial build error which seems fairly obviously broken in some fashion.   I believe it to be a version tangling which I point out below but that is hypothetical.   Is there a quick way to get unstuck here?

```
$ cat Cargo.toml
[package]
...

[dependencies]
ockam = "0"
```

Note below that both `ockam_core v0.7.0` and `ockam_core v0.6.0`.   That seems sub-optimal and possibly related to the error, but I do not have conclusive proof.

```
...
   Compiling ockam_core v0.7.0
   Compiling ockam_core v0.6.0                                                                                                       
   Compiling tracing-subscriber v0.2.17
   Compiling ockam_vault_core v0.3.2
   Compiling ockam_vault v0.3.2
   Compiling ockam_node v0.4.0
   Compiling ockam v0.4.1
error[E0277]: `?` couldn't convert the error to `ockam_core::Error`
   --> /home/joel/.cargo/registry/src/github.com-1ecc6299db9ec823/ockam-0.4.1/src/profile.rs:208:50
    |
208 |         let prev_id = v.sha256(Profile::NO_EVENT)?;
    |                                                  ^ the trait `std::convert::From<ockam_core::error::Error>` is not implemented for `ockam_core::Error`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following implementations were found:
              <ockam_core::Error as std::convert::From<Box<bincode::error::ErrorKind>>>
              <ockam_core::Error as std::convert::From<NodeError>>
              <ockam_core::Error as std::convert::From<credential::error::CredentialError>>
              <ockam_core::Error as std::convert::From<error::OckamError>>
              <ockam_core::Error as std::convert::From<ockam_node::error::Error>>
    = note: required by `std::convert::From::from`
```

**Comment by mrinalwadhwa** [comment]
@jbmohler thank you for reporting this 🙏

**Comment by jbmohler** [comment]
Digging into the examples, I have a work-around -- reference the github check-out directly from my sibling project directory:

```
$ cat Cargo.toml
[dependencies]
ockam = { version = "*", path = "../ockam/implementations/rust/ockam/ockam" }
```

**Comment by jared-s** [comment]
Hi @jbmohler, thank you SO much for reporting this. It has been fixed in `ockam v0.4.2`. If you build now, it should work. Really sorry about that!

**Comment by sheltont** [comment]
> Digging into the examples, I have a work-around -- reference the github check-out directly from my sibling project directory:
> 
> ```
> $ cat Cargo.toml
> [dependencies]
> ockam = { version = "*", path = "../ockam/implementations/rust/ockam/ockam" }
> ```

yes, it's a workaround.

**Comment by mrinalwadhwa** [comment]
Hi @sheltont are you still seeing the problem, even with the latest version?

**Comment by jbmohler** [comment]
This is fixed for me and I get a build with `0.4.2`.   My Cargo.toml says:

```
[dependencies]
ockam = "0"
ockam_transport_tcp = "0"
```
